src/helpers/getByAPI.js

```diff
@@ -1,24 +1,44 @@
 // @flow
 import * as React from 'react';
+import prettyFormat from 'pretty-format';
 import ErrorWithStack from './errorWithStack';
 
+const getNodeByType = (node, type) => node.type === type;
+
 const getNodeByName = (node, name) =>
-  node.type.name === name ||
-  node.type.displayName === name ||
-  node.type === name;
+  typeof node.type !== 'string' &&
+  (node.type.displayName === name || node.type.name === name);
+
+const getNodeByText = (node, text) => {
+  try {
+    // eslint-disable-next-line
+    const { Text, TextInput } = require('react-native');
+    return (
+      (getNodeByType(node, Text) || getNodeByType(node, TextInput)) &&
+      (typeof text === 'string'
```

Comment:
`typeof something === 'string'` is used couple of times throughout the code - can you extract it to method?

Reply:
Wish I could, but Flow looses type information because of fn extraction :<